.github/workflows/ci-docs-autogen.yml

```diff
@@ -0,0 +1,74 @@
+name: Autogenerated documentation
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  autogen-docs:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: yarn
+
+      - name: Install Foundry
+        uses: onbjerg/foundry-toolchain@v1
+        with:
+          version: nightly
+
+      - name: Node dependencies cache
+        uses: actions/cache@v3
+        with:
+          path: "node_modules"
+          key: yarn-${{ hashFiles('yarn.lock') }}
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+        shell: bash
+
+      - name: Generate docs
+        run: forge doc --build
+        shell: bash
+
+      - name: Upload docs
+        uses: actions/upload-artifact@v2
+        with:
+          name: docs-foundry
+          path: docs/book
+
+  upload-to-s3:
+    runs-on: ubuntu-latest
+    needs: autogen-docs
+    environment:
+      name: docs
+      url: https://developers.morpho.xyz
+    steps:
+      - name: Download docs
+        uses: actions/download-artifact@v2
+        with:
+          name: docs-foundry
+          path: docs
```

Comment:
Interested in the CI design: why did you separate the 2 jobs?

Reply:
There is multiple reason:
- I have first added a parallel step "upload to ipfs", removed because we are not using it
- We can have access to the artifacts if needed (if you need to access the code source)
- If we have an error in the upload (network error, credentials issue or concurrency in cloudfront invalidation), we can easily rerun the upload without having to regenerate the artifacts.